### DISCUSSION TITLE
Exclude surface unpaved

### DISCUSSION THREAD

**Discussion by lislainestachowoski** [Project-OSRM/osrm-backend#7052] [General]
Hi everyone!

Is it possible to exclude _surface_ _unpaved_ when I call endpoint OSRM with _exclude=unpaved_?
I already try adding unpaved on _excludable_, but doesn't work. 

I think I need a way to get this url parameter to filter unpaved surfaces on process_way function, but I dont know how to do this.

Can anyone help me ?

**Comment by jcoupey** [comment]
> I already try adding unpaved on excludable, but doesn't work.

The reason for that is that excludables apply to certain types of road (i.e. values of the OSM `highway` tag). You could adjust excludables to work out of the box with other existing values such as `highway=unclassified`. The thing with unpaved roads is that this is not the type of the road but an additional `unpaved=yes` tag so the filtering would not apply out of the box.

Do you want to be able to switch unpaved roads on and off on a query basis? Asking because if you want to simply remove them for good it would be simpler to filter them out in the profile.

**Comment by lislainestachowoski** [comment]
Hi @jcoupey, thanks for your answer.

Yes, I need to be able switch unpaved roads on and off on a query basis. 
Here in Brazil, we have many unpaved roads, and some trucks can't pass on them, but other types of vehicles can.
So we wanted to make it dynamic for each type of vehicle. 

I cant use unclassified, because the roads have classification, such as residential, or tertiary, but they are unpaved.

Here an example:
![image](https://github.com/user-attachments/assets/51457300-c127-4221-a5ee-ec068b0b9cae)


So, is there no way to do this filter for unpaved roads, using any parameter request, like exclude=unpaved?

**Comment by jcoupey** [comment]
> I cant use unclassified, because the roads have classification, such as residential, or tertiary, but they are unpaved.

Yes, that was my point: the current excludables logic does not apply in the case of an additional tag. I don't think you can do that without changing the code logic for excludables, at least in the lua profile. I have no idea how complex the adjustments would be...

**Comment by lislainestachowoski** [comment]
Ok @jcoupey !
Thanks for your help :)
